openapi.yml

```diff
@@ -45,26 +65,43 @@ paths:
           in: query
           schema:
             $ref: "#/components/schemas/Address"
+        - name: sellAmount
+          in: query
+          schema:
+            $ref: "#/components/schemas/TokenAmount"
+        - name: buyAmount
+          in: query
+          schema:
+            $ref: "#/components/schemas/TokenAmount"
+        - name: orderType
+          in: query
+          schema:
+            $ref: "#/components/schemas/Flags"
       responses:
         200:
-          description: the nonce
+          description: the tip
           content:
             application/json:
               schema:
-                $ref: "#/components/schemas/Nonce"
+                $ref: "#/components/schemas/Tip"
+        400:
+          description: sellToken or buyToken not existent
 components:
   schemas:
     Address:
       description: Ethereum 40 byte address encoded as a hex without `0x` prefix.
       type: string
       example: "6810e776880c02933d47db1b9fc05908e5386b96"
     TokenAmount:
-      description: Amount of a token. uint112 encoded in decimal.
+      description: Amount of a token. uint256 encoded in decimal.
       type: string
       example: "1234567890"
-    Nonce:
-      description: Nonce of the batch in which the order is valid. uint32.
+    Tip:
+      description: The tip that is used to pay solvers. uint256.
       type: integer
+    Flags:
+          description: Integer encodes the information for buy/sell order and fillOrKill/standard order uint8 encoded in decimal.
+          type: integer
```

Comment:
Why store the raw flags over the previous enum? To me this is internal to the encoding of the smart contract but for this api it makes more sense to keep the enums.

Reply:
I thought the parsing into enums is one additional "unrequited" step before verifying the signature. I order to create the digest, we need the flag-int anyways.

But probably, you are right that the interface is cleaner with orderType and fillType. Let me change it back.

Reply:
> but for this api it makes more sense to keep the enums

I also agree, especially considering how the signed data uses an `enum` and `bool` and the `u8` flags is just used for encoding for gas efficiency.